src/Typography/Typography.component.tsx

```diff
@@ -84,40 +86,52 @@ export interface TypographyProps extends SpaceProps, React.HTMLAttributes<any> {
 
 export interface StyledTypographyProps extends TypographyProps {
     $color?: 'inherit' | 'initial' | string;
+    asVariant?: any;
+    scaleVariant?: any;
 }
 
-export const Typography = styled('span').attrs<TypographyProps>(
-    ({ color }) => ({
-        $color: color,
-        color: undefined,
-        className: 'anchor-typography',
-    })
-)<StyledTypographyProps>`
+export const Typography = styled('span').attrs<TypographyProps>(props => ({
+    $color: props.color,
+    color: undefined,
+    className: 'anchor-typography',
+    asVariant: variant({
```

Comment:
This is so that we don't have to calculate the variant multiple times, now that its referenced in multiple places